go/tasks/plugins/array/awsbatch/client.go

```diff
@@ -68,22 +73,25 @@ func (b client) GetAccountID() string {
 }
 
 // Registers a new job definition. There is no deduping on AWS side (even for the same name).
-func (b *client) RegisterJobDefinition(ctx context.Context, name, image, role string) (arn definition2.JobDefinitionArn, err error) {
+func (b *client) RegisterJobDefinition(ctx context.Context, name, image, role string, structObj *structpb.Struct) (arn definition2.JobDefinitionArn, err error) {
 	logger.Infof(ctx, "Registering job definition with name [%v], image [%v], role [%v]", name, image, role)
+	jobDefinition := batch.RegisterJobDefinitionInput{}
+	if structObj != nil {
+		err = pluginUtils.UnmarshalStructToObj(structObj, &jobDefinition)
```

Comment:
what happens if structObj is nil? We will get null pointer exception on line 85?

Reply:
we should check if nil, then return an error?

Reply:
We create a `jobDefinition` object at line 78, so `jobDefinition` won't be nil at line 85. correct me if I'm wrong.

btw, check the test [here](https://github.com/flyteorg/flyteplugins/pull/228/files#diff-39aec2e3bb82fc78898943f508e5b70c528166f279c8a931f1cc37e20e6a48f0R92-R94)